### PR TITLE
add reasons for condition progressing

### PR DESCRIPTION
**Pull request by haoqing0110**

### CODE REVIEW

File: addon/v1alpha1/types_managedclusteraddon.go [hunk 1]
```diff
@@ -411,13 +407,37 @@ const (
...
+	// ProgressingReasonWaitingForCanary is the reason of condition Progressing indicating the addon configuration
+	// upgrade is pending and waiting for canary is done.
+	ProgressingReasonWaitingForCanary = "WaitingForCanary"
 
-	// AddonReasonConfigurationUnsupported is the reason of condition UnsupportedConfiguration indicating the configuration
-	// in clusterManagementAddon is not supported.
-	AddonReasonConfigurationUnsupported = "ConfigurationUnsupported"
+	// ProgressingReasongConfigurationUnsupported is the reason of condition Progressing indicating the addon configuration
+	// is not supported.
+	ProgressingReasongConfigurationUnsupported = "ConfigurationUnsupported"
```
Comment (qiujian16) [inline addon/v1alpha1/types_managedclusteraddon.go]:
```suggestion
	ProgressingReasonConfigurationUnsupported = "ConfigurationUnsupported"
```

Comment (haoqing0110) [reply addon/v1alpha1/types_managedclusteraddon.go]: fixed

**Pull request merged**